packages/flutter/test/material/navigation_bar_test.dart

```diff
@@ -553,6 +555,113 @@ void main() {
 
     expect(newHeight, equals(initialHeight));
   });
+
+  testWidgets('Navigation indicator renders ripple', (WidgetTester tester) async {
+    final Widget widget = _buildWidget(
+      NavigationBar(
+        destinations: const <Widget>[
+          NavigationDestination(
+            icon: Icon(Icons.ac_unit),
+            label: 'AC',
+          ),
+          NavigationDestination(
+            icon: Icon(Icons.access_alarm),
+            label: 'Alarm',
+          ),
+        ],
+        onDestinationSelected: (int i) {
+        },
+      ),
+    );
+
+    await tester.pumpWidget(widget);
+
+    await tester.tap(find.byIcon(Icons.access_alarm));
+    await tester.pump(const Duration(milliseconds: 200));
+
+    final RenderObject inkFeatures = tester.allRenderObjects.firstWhere((RenderObject object) => object.runtimeType.toString() == '_RenderInkFeatures');
+    expect(inkFeatures, paints..circle(color: const Color(0x66c8c8c8)));
+  });
+
+  testWidgets('Navigation indicator scale animation curve', (WidgetTester tester) async {
+    int selectedIndex = 0;
+
+    Widget buildNavigationBar() {
+      return MaterialApp(
+        theme: ThemeData.light(),
+        home: Scaffold(
+          bottomNavigationBar: Center(
+            child: NavigationBar(
+              selectedIndex: selectedIndex,
+              destinations: const <Widget>[
+                NavigationDestination(
+                  icon: Icon(Icons.ac_unit),
+                  label: 'AC',
+                ),
+                NavigationDestination(
+                  icon: Icon(Icons.access_alarm),
+                  label: 'Alarm',
+                ),
+              ],
+              onDestinationSelected: (int i) { },
+            ),
+          ),
+        ),
+      );
+    }
+
+    await tester.pumpWidget(buildNavigationBar());
+    await tester.pumpAndSettle();
+    final Finder transformFinder = find.descendant(
+      of: find.byType(NavigationIndicator),
+      matching: find.byType(Transform),
+    ).last;
+    Transform transformWidget = tester.widget(transformFinder);
+    expect(
+      transformWidget.transform,
+      Matrix4.diagonal3Values(
+        0.0,
+        1.0,
+        1.0,
+      ),
+    );
+
+    // Test scale animation with `Curves.fastOutSlowIn`.
+    selectedIndex = 1;
+    await tester.pumpWidget(buildNavigationBar());
+    await tester.pump(const Duration(milliseconds: 100));
+    transformWidget = tester.widget(transformFinder);
+    expect(
+      transformWidget.transform,
+      Matrix4.diagonal3Values(
+        0.4809186697006226,
```

Comment:
This constant, and the similar ones below, seem like a magnet for confusing FP stability issues. Maybe test one or three elements of of the transform matrix with closeTo.

Reply:
Great point, updated this.